cheta/fetch.py

```diff
@@ -190,6 +202,78 @@ def wrapper(*args, **kwargs):
     return the_decorator
 
 
+def _is_ephem_msid(msid: str) -> bool:
+    """True if this MSID is an ephemeris MSID
+
+    Parameters
+    ----------
+    msid : str
+        MSID to check.
+
+    Returns
+    -------
+    bool
+        True if this MSID is an ephemeris MSID.
+    """
+    return content.get(msid.upper()) in EPHEM_CONTENT_TYPES
+
+
+def _get_ephem_times_ok(times: np.ndarray[float]) -> np.ndarray[bool]:
+    """Select most recent sample times for ephemeris data from overlapping chunks.
+
+    The raw ephemeris HDF5 archive files are non-monotonic because the archive FITS
+    files are overlapping and have duplicate times. Typically new ephemeris files come
+    out each 7 days and cover many weeks of data (e.g. about 10 weeks for the predictive
+    EPHEM 0 files).
+
+    The HDF5 files on HEAD have TIME.quality set to bad for the duplicate times, so this
+    code is not strictly required there. But for archives maintained with cheta_sync,
+    the quality codes are not back-updated when new data appear, causing problems
+    (https://github.com/sot/cheta/issues/245).
+
+    This function ignores the existing quality entirely and dynamically assembles a mask
+    of the best (latest) time sample to use.
+
+    Parameters
+    ----------
+    times : np.ndarray[float]
+        Array of time samples (seconds since 1998.0) from ephemeris MSID.
+
+    Returns
+    -------
+    np.ndarray[bool]
+        Boolean mask array that is True for the best samples to keep.
+    """
+    # Baseline times to start at exactly zero and round to nearest 10. Ephem are
+    # nominally sampled at exactly 300 sec intervals, but sometimes there are floating
+    # point variations and sometimes 301 (leap second?), and at least one larger gap of
+    # 43436 seconds.
+    times = np.round(times - times[0], -1)
+
+    # Times here come from complete and overlapping archive file chunks. This means that
+    # each chunk starts in the middle of the previous chunk and extends beyond. First
+    # get the index of the start of each chunk where time jumps backward.
+    dts = np.diff(times)
+    idx_neg_jumps = np.where(dts < 0)[0] + 1
+
+    # Get complete chunks as pairs of (idx_start, idx_stop)
+    idx_chunks = np.concatenate([[0], idx_neg_jumps, [len(times)]])
+    chunks = list(itertools.pairwise(idx_chunks))
+
+    # Make the mask by masking out every time in the previous chunk that is after the
+    # start of the current chunk. This mirrors what is done in archive update processing.
+    # This process works for fetching data on the primary HEAD archive, but does not
+    # work for cheta_sync archives because the quality flags are back-updated in the
+    # primary archive but not in cheta_sync.
```

Comment:
I'm a bit confused here about what "This process" is and what isn't going to work with this new code.